src/pkg/runner/runner.go

```diff
@@ -175,6 +187,63 @@ func (r *Runner) importTasks(includes []map[string]string, dir string, setVariab
 	return nil
 }
 
+func (r *Runner) loadIncludedTaskFile(taskName string) (string, error) {
+	// Check if running task directly from included task file
+	includedTask := strings.Split(taskName, ":")
+	if len(includedTask) == 2 {
+		includeName := includedTask[0]
+		includeTaskName := includedTask[1]
+		var fullPath string
+		var tasksFile types.TasksFile
+		templatePattern := `\${[^}]+}`
+		re := regexp.MustCompile(templatePattern)
+		// Get referenced include file
+		for _, includes := range r.TasksFile.Includes {
+			// Check if include exists
+			if includeFileLocation, ok := includes[includeName]; ok {
+				// check for templated variables in includeFileLocation value
+				if re.MatchString(includeFileLocation) {
+					r.PopulateTemplateMap(r.TasksFile.Variables, config.SetRunnerVariables)
+					includeFileLocation = r.TemplateString(includeFileLocation)
+				}
+				// check if included file is a url
+				if helpers.IsURL(includeFileLocation) {
+					// If file is a url download it to a tmp directory
+					tmpDir, err := zarfUtils.MakeTempDir(zarfConfig.CommonOptions.TempDirectory)
```

Comment:
same comment as before regarding downloading directly into a struct instead of making a tmp dir

Reply:
Maybe this doesn't work here because of the `config.TaskFileLocation`....

Reply:
yeah, config.TaskFileLocation gets used globally in other areas so it will need to get updated with that temp file location